compiler/parser/parser.cpp

```diff
@@ -330,6 +330,14 @@ static void parseCommandLineFiles() {
   while ((inputFileName = nthFilename(fileNum++))) {
     if (isChplSource(inputFileName)) {
       parseFile(inputFileName, MOD_USER, true, false);
+      /*
+      Ensure that all the files parsed don't exceed the
+      NAME_MAX/2 125 bytes Limits
+      */
+      if (strlen(inputFileName) > NAME_MAX / 2 - 16)
+      {
+        USR_FATAL("File name too long");
```

Comment:
Hi @king-11 : Thanks for tackling this.  I'm not familiar with NAME_MAX, but was just googling it.  Is the idea that FILENAME_MAX limits the size of `/full/path/to/filename.chpl` whereas `NAME_MAX` says how long each of these components (like `filename.chpl` specifically) can be?

What's the rationale for using NAME_MAX / 2 -16?  I think this should be rationalized in the source comment.

I'd be inclined to move this conditional before the parseFile() call (why bother parsing it, if we're just going to generate an error about the filename length?  E.g., it'd be frustrating to work through a number of syntax errors and then be told that the filename was too long.

I think it'd be good if the error message printed the filename for reference, and also what the maximum length is so the user knows how much shorter they need to make it.  I'd also probably use "Filename"  rather than "File name" in the error message.

Can you add a test to the test system, similar to the one in the original issue to show that this error is triggered appropriately and that the behavior is locked in?

Reply:
- Yes the limits for `complete path` length I think are long enough in every OS so won't hinder any working. I encountered these (path limits) while working on limits for just the `filename` so that's what FILENAME_MAX ensures
- the `NAME_MAX/2` was based on the part that `strlen` resulted in char count but saw your comment so that won't be needed anymore. the `-16` was just a random choice in terms of `2's` power while `254` is also in the limits of OS like `Mac`, `Linux` and `Windows` I added a decrease of `16` for other OS also
- I will change the location for error Throwing, add tests and make the error message more precise/understandable